server/service/microsoft_mdm.go

```diff
@@ -621,6 +547,30 @@ func (svc *Service) GetMDMMicrosoftDiscoveryResponse(ctx context.Context) (*flee
 	return &discoveryMsg, nil
 }
 
+// GetMDMMicrosoftPolicyResponse returns a valid GetPoliciesResponse message
+func (svc *Service) GetMDMMicrosoftPolicyResponse(ctx context.Context, authToken string) (*fleet.GetPoliciesResponse, error) {
+	if len(authToken) == 0 {
+		return nil, ctxerr.Wrap(ctx, errors.New("invalid parameters"), "authToken is empty")
```

Comment:
We have an error type for invalid parameters, `fleet.InvalidArgumentError`, you can see it in use e.g. in `ModifyAppConfig`, I think we should still use it for Windows MDM even though we end up returning a SOAP error and not an HTTP one (unless there's some technical issue using it?), as it will look like something that should be fixed by future readers. The service should not care whether its result will be SOAP or HTTP, the error should be semantically the right type.

Reply:
Thanks for the suggestion! I've just addressed this one. I agree that `fleet.InvalidArgumentError` should be used here